quest_test/test_persistent_history.py

```diff
@@ -0,0 +1,87 @@
+from src.quest.persistence import InMemoryBlobStorage, ListPersistentHistory, LinkedPersistentHistory
+from src.quest.types import StepStartRecord, ResourceEntry, TaskEvent, EventRecord
+from datetime import datetime, UTC
+
+step_rec = StepStartRecord()
+step_rec['type'] = 'start'; step_rec['timestamp'] = datetime.now(UTC).isoformat()
+step_rec['step_id'] = 'record_1_identifier'; step_rec['task_id'] = 'record_1_task'
+
+step_rec2 = StepStartRecord()
+step_rec2['type'] = 'start'; step_rec2['timestamp'] = datetime.now(UTC).isoformat()
+step_rec2['step_id'] = 'record_2_identifier'; step_rec2['task_id'] = 'record_2_task'
+
+task_rec = TaskEvent()
+task_rec['type'] = 'start_task'; task_rec['timestamp'] = datetime.now(UTC).isoformat()
+task_rec['step_id'] = 'record_3_identifier'; task_rec['task_id'] = 'record_3_task'
+
+records: list[EventRecord] = [step_rec, step_rec2, task_rec]
+
+def test_linked_history():
+    storage = InMemoryBlobStorage()
+    linked_history = LinkedPersistentHistory('linked_history', storage)
```

Comment:
rename to `history` and parameterize this function to take the type of PersistentHistory being tested (i.e. `LinkedPersistentHistory` vs `ListPersistentHistory`)

Use `'history'` instead of `'linked_history'`

Reply:
then delete the extra test below